tests/phpunit/tests/block-template.php

```diff
@@ -275,6 +275,89 @@ public function test_get_the_block_template_html_does_not_generally_enforce_loop
 		$this->assertSame( array( false, true ), $in_the_loop_logs, 'Main query loop was triggered incorrectly' );
 	}
 
+	/**
+	 * @ticket 58319
+	 *
+	 * @covers ::get_block_theme_folders
+	 *
+	 * @dataProvider data_get_block_theme_folders
+	 *
+	 * @param string   $theme    The theme's stylesheet.
+	 * @param string[] $expected The expected associative array of block theme folders.
+	 * @param string   $message The message for when an assertion fails.
+	 */
+	public function test_get_block_theme_folders( $theme, $expected, $message ) {
+		$wp_theme   = wp_get_theme( $theme );
+		$reflection = new ReflectionMethod( $wp_theme, 'cache_get' );
+		$reflection->setAccessible( true );
+		$wp_theme->cache_delete(); // Clear cache.
+
+		$this->assertSame( $expected, get_block_theme_folders( $theme ), $message );
+		$theme_cache  = $reflection->invoke( $wp_theme, 'theme' );
+		$cached_value = $theme_cache['block_template_folders'];
+		$reflection->setAccessible( false );
```

Comment:
```suggestion
	 */
	public function test_get_block_theme_folders( $theme, $expected, $message ) {
		$wp_theme = wp_get_theme( $theme );
		$wp_theme->cache_delete(); // Clear cache.

		$this->assertSame( $expected, get_block_theme_folders( $theme ), $message );
		$reflection = new ReflectionMethod( $wp_theme, 'cache_get' );
		$reflection->setAccessible( true );
		$theme_cache  = $reflection->invoke( $wp_theme, 'theme' );
		$cached_value = $theme_cache['block_template_folders'];
		$reflection->setAccessible( false );
```
Moving the reflection and access changes until after the first assertion in case of assertion failure.